### DISCUSSION TITLE
How can i get mask from Inet field type

### DISCUSSION THREAD

**Discussion by RrredHead** [prisma/prisma#17903] [Q&A] [answered]
I have a project with table configured like that

```
generator client {
    provider        = "prisma-client-js"
    previewFeatures = ["metrics"]
}

datasource db {
    provider = "postgresql"
    url      = env("DATABASE_URL")
}

// ...

model IP {
    id          Int      @id @default(autoincrement())
    address     String   @unique @db.Inet
    description String   @default("")
    updatedAt   DateTime @updatedAt
}
```

Which gives me table as I expected (DDL as example): 

```sql
CREATE TABLE public."IP" (
	id serial4 NOT NULL,
	description text NOT NULL DEFAULT ''::text,
	"updatedAt" timestamp(3) NOT NULL,
	address inet NOT NULL,
	CONSTRAINT "IP_pkey" PRIMARY KEY (id)
);
CREATE UNIQUE INDEX "IP_address_key" ON public."IP" USING btree (address);
```

In this table I have only one record (for test purposes):

<img width="463" alt="image" src="https://user-images.githubusercontent.com/5057549/218415851-b0049636-eb8c-4238-b04a-3298ac9ec790.png">

Also I have a route, which should just return every records stored in IP table

```typescript
const ips = await prisma.iP.findMany();
return ips;
```

As a result it gives:

```json
[
  {
    "id": 2,
    "description": "Localhost",
    "updatedAt": "2023-02-13T14:05:04.121Z",
    "address": "127.0.0.0"
  }
]
```

Everything is nearly correct, but there is no mask in address field. In this case I'm expecting "address": "127.0.0.0/16" as it stored in Data Base. How can i achieve the expected behavior?

**Comment by ludralph** [comment]
Hi @RrredHead 👋 

This looks like the default behavior of the database. You can see a reference [here](https://www.postgresql.org/docs/current/datatype-net-types.html#:~:text=The%20inet%20type%20holds%20an,subnet%2C%20only%20a%20single%20host)

If you want to include the mask, you might want to consider using the [net functions](https://www.postgresql.org/docs/8.2/functions-net.html), but then you would need to utilize [raw queries](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) within Prisma.

**Comment by RrredHead** [comment]
Hi @ludralph!

Here are some quotes from this section of reference

> If the netmask is 32 and the address is IPv4, then the value does not indicate a subnet, only a single host.

and

> On display, the /y portion is suppressed if the netmask specifies a single host.

But as you can see in my example, the netmask is `/16`, so it should be in the output, if no functions applied to `address` field. 

Actually it looks like function `host` applied under the hood of prisma (look at [Table 9-32 here](https://www.postgresql.org/docs/8.2/functions-net.html) in reference).

**Comment by RrredHead** [comment, marked as answer]
OK, I've found one workaround:

```typescript
async function findIPs() {
    const result = await prisma.$queryRaw<
        IP[]
    >`SELECT id, description, "updatedAt", text(address) as address FROM public."IP"`;
    return result;
}

// ... later in route handler

const ips = await findIPs();
return ips;
```

This gives me output as I expected:

```json
[
  {
    "id": 2,
    "description": "Localhost",
    "updatedAt": "2023-02-13T14:05:04.121Z",
    "address": "127.0.0.0/16"
  }
]
```

But I'm quite sure that function `text` have to be applied when casting postgres INET datatype to JS String by default 🤔

**Reply by ludralph** [reply]
I do not know the implementation details, but it looks like the host was used under the hood. So the workaround you've provided looks good.

**Comment by RrredHead** [comment]
I've created an issue from this discussions: https://github.com/prisma/prisma/issues/17929

**Reply by ludralph** [reply]
Thank you for creating an issue. A member of the engineering team will take a look at it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
